Document restricted domains and host permissions (#26188)
mdn/content: files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md

Content scripts can only access [a small subset of the WebExtension APIs](#webextension_apis), but they can [communicate with background scripts](#communicating_with_background_scripts) using a messaging system, and thereby indirectly access the WebExtension APIs.

> **Note:** Content scripts are only executed if the extension is granted [host permissions](/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions) for the domain.
>

> > [Restricted domains](#restricted_domains) can never be accessed, regardless of the granted permissions.

Reviewer: ```suggestion
> However, [restricted domains](#restricted_domains) can never be accessed, regardless of the granted permissions.
```

> Only URLs with the _moz-extension_ scheme are allowed, which excludes data URLs ([Firefox bug 1587336](https://bugzil.la/1587336)).

## Restricted domains

Even with the right [host permissions](/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions), some domains cannot be accessed by extensions. Content scripts are blocked from executing on these domains, for example to protect the user from an extension escalating privileges through special pages.

In Firefox, this includes the following domains:

- accounts-static.cdn.mozilla.net
- accounts.firefox.com
- addons.cdn.mozilla.net
- addons.mozilla.org
- api.accounts.firefox.com
- content.cdn.mozilla.net
- discovery.addons.mozilla.org
- install.mozilla.org
- oauth.accounts.firefox.com
- profile.accounts.firefox.com
- support.mozilla.org
- sync.services.mozilla.com

Other browsers have similar restrictions for the website from where extensions can be installed. For example, access to chrome.google.com is restricted in Chrome.

> Because these restrictions include addons.mozilla.org, users may attempt to use your extension immediately after installation—only to find that it doesn't work! You may want to add an appropriate warning, or an [onboarding page](https://extensionworkshop.com/documentation/develop/onboard-upboard-offboard-users/) to move users away from `addons.mozilla.org`.

Reviewer: ```suggestion
> **Warning:** Because these restrictions include addons.mozilla.org, users who try to use your extension immediately after installation may find that it doesn't work. To avoid this, you should add an appropriate warning or an [onboarding page](https://extensionworkshop.com/documentation/develop/onboard-upboard-offboard-users/) to move users away from `addons.mozilla.org`.
```
